### PR TITLE
On-demand KeyExchange

### PR DESCRIPTION
**Pull request by toly-kournik**
- CryptoManager to support multiple crypto systems accessed by a seqnum;
- Hence removed key rotation logic. Instead old keys are garbage collected on checkpoint;
- ExecutionRequest to include CID;
- ISecureStore + hierarchiy deleted in favour of existing ISecretsManagerImpl;
- KeyExchangeMgr
-- Moved to include/bftengine as it needs to be accessed by other components;
-- Holds replica's private keys collection per seqnum, as well as newly generated key pair;
-- Initialization logic changed to use RAII;
- Serializable to has a default impl of getVersion() in order not to implement it subclasses if not needed;

### CODE REVIEW

File: bftengine/tests/keyManager/CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
-    GTest::Main
-    corebft)
+#find_package(GTest REQUIRED)
+#
+#add_executable(KeyManager_test KeyManager_test.cpp )
+#add_test(KeyManager_test KeyManager_test)
+#
+#target_link_libraries(KeyManager_test PUBLIC
+#    GTest::Main
+#    corebft)
```
Comment (yontyon) [inline bftengine/tests/keyManager/CMakeLists.txt:8]: Why not removing this?

**Review by yontyon — APPROVED** [review]
LGTM!

**Review by cloudnoize — APPROVED** [review]
Looks good thanks @toly-kournik

File: bftengine/tests/keyManager/CMakeLists.txt (hunk 1 above)
Comment (toly-kournik) [reply bftengine/tests/keyManager/CMakeLists.txt:8]: I still need to change the unittest

**Pull request merged**